### PR TITLE
add device to DB

### PR DESCRIPTION
**Pull request by Efrat6821**

### CODE REVIEW

File: lib/model/src/Device.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+interface Model {
+    device_id: string;
+    SIM_number: number;
+    IMEI_1: number;
+    mehalcha_number: number;
```
Comment (klakdev) [inline lib/model/src/Device.ts:7]: What does this mean? is it Hebrew? 
Only use English names for variables.

File: lib/backend/src/controller/device.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            };
+        const deleteDevice = await db.Device.deleteDevice(req.params.id);
+        res.status(200).json(deleteDevice);
+    } catch (error: any) {
+        handleError(res, error);
+    }
+};
+
+const handleError = (res: Response, error: any): void => {
+    console.error('Error:', error.message);
```
Comment (klakdev) [inline lib/backend/src/controller/device.ts]: Share this between device and customer,

File: lib/backend/src/controller/device.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    } catch (error: any) {
+        handleError(res, error);
+    }
+};
+
+const deleteDevice = async (req: Request, res: Response): Promise<void> => {
+    try {
+        if (!req.params.id)
+            throw {
+                status: 400,
```
Comment (klakdev) [inline lib/backend/src/controller/device.ts]: This should be checked with the DB.

select exists id from device where id = $1

File: lib/backend/src/controller/device.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+const updateDevice = async (req: Request, res: Response): Promise<void> => {
+    try {
+        if (!req.body)
+            throw {
+                status: 400,
+                message: 'No body provaider'
+            }
+        const sanitized = Device.sanitize(req.body, true);
+        const devices: Device.Model[] = await db.Device.getDevices();
```
Comment (klakdev) [inline lib/backend/src/controller/device.ts]: find the device in the db
selct * from device where id = $1 or imei = $2...

File: lib/backend/src/controller/device.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+const getDeviceById = async (req: Request, res: Response): Promise<void> => {
+    try {
+        if (!req.params.id)
+            throw {
+                status: 400,
+                message: 'No ID provided'
+            };
+        const device = await db.Device.getDeviceById(req.params.id);
+        if (!device) {
```
Comment (klakdev) [inline lib/backend/src/controller/device.ts]: just call next
let express handle 404

File: lib/backend/src/controller/device.ts [hunk 6]
```diff
@@ -0,0 +1,127 @@
+import { Request, Response } from 'express';
+import db from "../db";
+import { Device } from "../model";
+
+const createDevice = async (req: Request, res: Response): Promise<void> => {
+    try {
+        const deviceData = req.body;
+        const sanitized = Device.sanitize(deviceData, false);
+        const devices: Device.Model[] = await db.Device.getDevices();
```
Comment (klakdev) [inline lib/backend/src/controller/device.ts]: 1. see comment below.
2. do not duplicate code

**Review by klakdev — CHANGES_REQUESTED** [review]

File: lib/backend/src/controller/device.ts (hunk 5 above)
Comment (Efrat6821) [reply lib/backend/src/controller/device.ts]: לשנות את כל הזריקות שגיאה שכבר זרקתי???

**Pull request merged**